src/main/java/com/efficio/fieldbook/web/trial/controller/BaseTrialController.java

```diff
@@ -454,75 +454,65 @@ TabInfo prepareEnvironmentsTabInfo(final Workbook workbook, final boolean isUseP
 
 		info.setSettingMap(settingMap);
 
-		final EnvironmentData data = new EnvironmentData();
+		final InstanceInfo data = new InstanceInfo();
 		final List<MeasurementRow> trialObservations = workbook.getTrialObservations();
 
-		data.setNoOfEnvironments(trialObservations.size());
+		data.setNumberOfInstances(trialObservations.size());
 
-		final List<Environment> environments = new ArrayList<>();
+		final List<Instance> instances = new ArrayList<>();
 		for (final MeasurementRow row : trialObservations) {
-			final Environment environment = new Environment();
+			final Instance instance = new Instance();
 			if (!isUsePrevious) {
-				environment.setExperimentId(row.getExperimentId());
-				environment.setLocationId(row.getLocationId());
-				environment.setStockId(row.getStockId());
+				instance.setInstanceId(row.getLocationId());
+				instance.setStockId(row.getStockId());
 			}
 
 			final Map<String, String> managementDetailValues = new HashMap<>();
+			final Map<String, String> trialConditionValues = new HashMap<>();
+			final Map<String, Integer> phenotypeIDMap = new HashMap<>();
+			final Map<String, Integer> instanceDataIdMap = new HashMap<>();
 			for (final SettingDetail detail : managementDetailList) {
 
 				final MeasurementData mData = row.getMeasurementData(detail.getVariable().getCvTermId());
 				if (mData != null) {
 					final String value;
-					if ("DATE".equals(detail.getVariable().getWidgetType().getType())) {
```

Comment:
Why is the logic for DATE variable removed?

Reply:
I removed it because the system no longer needs to convert the date to the frontend format. I adjusted the code in the frontend to properly handle the date in numeric format.